cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/TransformExecutorFactory.java

```diff
@@ -40,14 +41,18 @@
  * @param <T> the type of input for the created transform executors
  */
 public abstract class TransformExecutorFactory<T> {
-  private final String sourceStageName;
+  protected final String sourceStageName;
+  protected final Map<String, Map<String, Schema>> perStageInputSchemas;
   protected final PipelinePluginInstantiator pluginInstantiator;
   protected final Metrics metrics;
+  protected Schema outputSchema;
 
   public TransformExecutorFactory(PipelinePluginInstantiator pluginInstantiator, Metrics metrics,
                                   @Nullable String sourceStageName) {
     this.pluginInstantiator = pluginInstantiator;
     this.metrics = metrics;
+    this.perStageInputSchemas = new HashMap<>();
```

Comment:
This is used to create [runtime context](https://github.com/caskdata/cdap/pull/5914/files#diff-5dd7976c5ba9e968cab1c7d1c79721cbR99) for joiner which exposes input schemas and output schema to the plugin.